packages/expo-router/src/ui/NativeBottomTabs/NativeTabsView.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  DefaultRouterOptions,
+  ParamListBase,
+  TabNavigationState,
+  TabRouterOptions,
+  useNavigationBuilder,
+} from '@react-navigation/native';
+import React from 'react';
+import { type ColorValue, type TextStyle } from 'react-native';
+import {
+  BottomTabs,
+  BottomTabsScreen,
+  enableFreeze,
+  featureFlags,
+  type BottomTabsProps,
+  type BottomTabsScreenProps,
+  type TabBarItemLabelVisibilityMode,
+} from 'react-native-screens';
+
+import { shouldTabBeVisible } from './utils';
+import { getPathFromState } from '../../link/linking';
+
+// We let native tabs to control the changes. This requires freeze to be disabled for tab bar.
+// Otherwise user may see glitches when switching between tabs.
+featureFlags.experiment.controlledBottomTabs = false;
+
+export type NativeTabOptions = Omit<
+  BottomTabsScreenProps,
+  | 'children'
+  | 'placeholder'
+  | 'onWillAppear'
+  | 'onDidAppear'
+  | 'onWillDisappear'
+  | 'onDidDisappear'
+  | 'isFocused'
+  | 'tabKey'
+> &
+  DefaultRouterOptions & { hidden?: boolean };
+
+export interface NativeTabsViewProps {
+  style?: {
+    fontFamily?: TextStyle['fontFamily'];
+    fontSize?: TextStyle['fontSize'];
+    fontWeight?: TextStyle['fontWeight'];
+    fontStyle?: TextStyle['fontStyle'];
+    color?: TextStyle['color'];
+    iconColor?: ColorValue;
+    backgroundColor?: ColorValue;
+    blurEffect?: BottomTabsScreenProps['tabBarBlurEffect'];
+    tintColor?: ColorValue;
+    badgeBackgroundColor?: ColorValue;
+    /**
+     * @platform android
+     */
+    rippleColor?: ColorValue;
+    /**
+     * @platform android
+     */
+    labelVisibilityMode?: TabBarItemLabelVisibilityMode;
+    '&:active'?: {
```

Comment:
The subclass would lead me to think other subclass styles should exist such as for icon and badge